pkg/scheduler_extender.go

```diff
@@ -32,10 +38,44 @@ const (
 type SchedulerExtender struct {
 	Clientset kubernetes.Interface
 	*logrus.Entry
+	Validator     validator.Validator
+	EventRecorder eventing.Recorder
 }
 
 // Update updates csi-baremetal-se or creates if not found
 func (n *SchedulerExtender) Update(ctx context.Context, csi *csibaremetalv1.Deployment, scheme *runtime.Scheme) error {
+	// in case of Openshift deployment and non default namespace - validate extender service accounts security bindings
+	if csi.Spec.Platform == constant.PlatformOpenShift && csi.Namespace != constant.DefaultNamespace {
+		var rbacError rbac.Error
+		if err := n.Validator.ValidateRBAC(ctx, &models.RBACRules{
+			Data: &rbacmodels.ServiceAccountIsRoleBoundData{
+				ServiceAccountName: constant.ExtenderServiceAccountName,
+				Namespace:          csi.Namespace,
+				Role: &rbacv1.Role{
+					Rules: []rbacv1.PolicyRule{
+						{
+							Verbs:         []string{"use"},
+							APIGroups:     []string{"security.openshift.io"},
+							Resources:     []string{"securitycontextconstraints"},
+							ResourceNames: []string{"privileged"},
+						},
+					},
+				},
+			},
+			Type: models.ServiceAccountIsRoleBound,
+		}); err != nil {
+			if errors.As(err, &rbacError) {
+				n.EventRecorder.Eventf(ctx, csi, eventModels.WarningType, "ExtenderRoleValidationFailed",
```

Comment:
Failed to validate or account is not privileged?

Reply:
fix comment to 'ServiceAccount csi-node-sa has insufficient securityContextConstraints, should have privileged'